Patch that fixes Debian bug #391681.

This patch by Josh Triplett.
Submitted to Mesa and applied (post 6.5.1).

--- a/src/glx/x11/glxext.c
+++ b/src/glx/x11/glxext.c
@@ -1016,11 +1016,11 @@
 	if (!_XReply(dpy, (xReply*) &reply, 0, False)) {
 	    /* Something is busted. Punt. */
 	    UnlockDisplay(dpy);
+	    SyncHandle();
 	    FreeScreenConfigs(priv);
 	    return GL_FALSE;
 	}
 
-	UnlockDisplay(dpy);
 	if (!reply.numVisuals) {
 	    /* This screen does not support GL rendering */
 	    UnlockDisplay(dpy);
